### PR TITLE
Simplify function definitions

### PR DESCRIPTION
**Pull request by jackfirth**
This is an automated change generated by [`resyntax`](https://github.com/jackfirth/resyntax).

This change simplifies function definitions using shortcuts for `define` forms whose right-hand-side is either a lambda expression or a `case-lambda` expression equivalent to a function with a default argument.

### CODE REVIEW

File: racket/collects/ffi/unsafe/define.rkt [hunk 1]
```diff
@@ -8,21 +8,19 @@
...
-                 (apply
-                  string-append
-                  "arguments:"
-                  (let loop ([args args])
-                    (if (null? args)
-                        null
-                        (cons (format " ~e"
-                                      (car args))
-                              (loop (cdr args)))))))))))
+(define (((make-not-available id)) . args)
```
Comment (samth) [inline racket/collects/ffi/unsafe/define.rkt:11]: This might be less clear in the new version -- maybe 0-argument functions should be a special case?

**Review by samth — COMMENTED** [review]
This broadly looks good to me but it changes a lot of code so I'm not sure who the right person to review it is.

File: racket/collects/ffi/unsafe/define.rkt (hunk 1 above)
Comment (jackfirth) [reply racket/collects/ffi/unsafe/define.rkt:11]: I think that makes sense. Opened https://github.com/jackfirth/resyntax/issues/61 for this.

**Comment by jackfirth** [comment]
I could split it up into separate pull requests, would that be easier?

**Comment by rfindler** [comment]
I looked through all of the contract files and the changes seem good to me. I didn't check that the test suites still pass (although it seems very likely they would). If someone has it installed, running `racket -l tests/racket/contract/all` runs them.

**Comment by samth** [comment]
The contract tests run in the CI so we can see that they pass.

**Comment by jackfirth** [comment]
Closing this pull request for a few reasons:

1. It's too large. I think these kinds of pull requests should probably be in the range of a couple hundred lines instead, at least while Resyntax is still in such an experimental state.
1. I'd like to fix jackfirth/resyntax#61 first.
1. It's been a while and now there's merge conflicts.

**Pull request closed without merging**